src/main/java/org/sigmah/shared/dto/element/TextAreaElementDTO.java

```
@@ -87,471 +89,486 @@ protected void appendToString(final ToStringBuilder builder) {
+		} else {
+			Log.warn("No textarea type is specified for the textarea element '" + getLabel() + "'. Using paragraph instead.");
+			field = createParagraphField(valueResult);
+		}
 
-					// Sets the min value.
-					final Long minValue;
-					if (getMinValue() != null) {
-						minValue = getMinValue();
-						numberField.setMinValue(minValue);
-					} else {
-						minValue = null;
-					}
+		// Sets the global properties.
+		field.setAllowBlank(true);
+		field.setFieldLabel(getLabel());
 
-					// Sets the min value.
-					final Long maxValue;
-					if (getMaxValue() != null) {
-						maxValue = getMaxValue();
-						numberField.setMaxValue(maxValue);
-					} else {
-						maxValue = null;
-					}
+		field.setEnabled(enabled);
 
-					// Sets tooltip.
-					numberField.setToolTip(I18N.MESSAGES.flexibleElementTextAreaNumberRange(
-						isDecimal ? I18N.CONSTANTS.flexibleElementDecimalValue() : I18N.CONSTANTS.flexibleElementNonDecimalValue(),
-						minValue != null ? String.valueOf(minValue) : "-", maxValue != null ? String.valueOf(maxValue) : "-"));
+		return field;
+	}
 
-					// Adds listeners.
-					numberField.addListener(Events.OnKeyUp, new Listener<BaseEvent>() {
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public boolean isCorrectRequiredValue(ValueResult result) {
 
-						@Override
-						public void handleEvent(BaseEvent be) {
+		if (result == null || !result.isValueDefined()) {
+			return false;
+		}
 
-							final Number number = numberField.getValue();
-							final Double asDouble = number != null ? number.doubleValue() : null;
+		final String value = result.getValueObject();
+		final boolean correct;
+		
+		final TextAreaType type = TextAreaType.fromCode(getType());
+		if (type == TextAreaType.DATE) {
+			correct = isCorrectRequiredDateValue(value);
+		} else if (type == TextAreaType.NUMBER) {
+			correct = isCorrectRequiredNumberValue(value);
+		} else {
+			correct = isCorrectRequiredStringValue(value);
+		}
```

You could `throw new UnsupportedOperationException("Given type '" + type + "' is not supported yet.");` just like you did above. That's pretty clear.